virttest/qemu_storage.py

```diff
@@ -1554,11 +1554,31 @@ def measure(self, target_fmt, size=None, output="human"):
         """
         cmd_list = [self.image_cmd, "measure", ("--output=%s" % output),
                     ("-O %s" % target_fmt)]
+
+        if target_fmt == "luks":
+            target_image = self.params.get("image_measure_target", "tgt")
+            target_image_params = self.params.object_params(target_image)
+            target_image_params["image_format_%s" % target_image] = "luks"
```

Comment:
do we need target_image here and next line? we get the image params by object_params, then target_image_params holds all params specified for the target_image itself, so here we might only need
           target_image_params["image_format"] = "luks"

we now can get the correct result  by these settings because in retrieve_secrets, we have to call params.object_params again for each image in image_chain...

Reply:
Agree with you, I will update it. Many thanks.